Mage.Sets/src/mage/cards/l/Lifeline.java

```diff
@@ -0,0 +1,146 @@
+ *
+ * @author HCrescent
+ */
+public class Lifeline extends CardImpl {
+    
+private static final FilterCreaturePermanent filter = new FilterCreaturePermanent("creature");
+
+    public Lifeline(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ARTIFACT}, "{5}");
+                
+
+        // Whenever a creature dies, if another creature is on the battlefield, return the first card to the battlefield under its owner's control at the beginning of the next end step.
+        Ability ability = new ConditionalTriggeredAbility(
+                            new DiesCreatureTriggeredAbility(
+                                new LifelineEffect(),
+                                false,
+                                false,
+                                true),
+                            new PermanentsOnTheBattlefieldCondition(filter, PermanentsOnTheBattlefieldCondition.CountType.MORE_THAN, 0, false),
+                            "Whenever a creature dies, if another creature is on the battlefield, return the first card to the battlefield under its owner's control at the beginning of the next end step.");
+        this.addAbility(ability);
+    }
+
+    public Lifeline(final Lifeline card) {
+        super(card);
+    }
+
+    @Override
+    public Lifeline copy() {
+        return new Lifeline(this);
+    }
+}
+class LifelineEffect extends OneShotEffect {
+    
+    public LifelineEffect() {
+        super(Outcome.Benefit);
+        this.staticText = "";
+    }
+    
+    public LifelineEffect(final LifelineEffect effect) {
+        super(effect);
+    }
+    
+    @Override
+    public LifelineEffect copy() {
+        return new LifelineEffect(this);
+    }
+    
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Cards cards = new CardsImpl();
+        Permanent permanent = game.getPermanentOrLKIBattlefield(this.getTargetPointer().getFirst(game, source));
+        if (permanent != null) {
+            //Tracking zone changes after first effect
+            permanent.setZoneChangeCounter(0, game);
+            cards.add(permanent);
+        }
+        Effect effect = new LifelineDelayedEffect();
+        effect.setValue("LifelineCards", cards);
+        game.addDelayedTriggeredAbility(new AtTheBeginOfNextEndStepDelayedTriggeredAbility(effect), source);
+        return true;
+    }
+}
+class LifelineDelayedEffect extends OneShotEffect {
+    
+    public LifelineDelayedEffect() {
+        super(Outcome.Benefit);
+        this.staticText = "return this card to the battlefield under it's owner's control.";
+    }
+    
+    public LifelineDelayedEffect(final LifelineDelayedEffect effect) {
+        super(effect);
+    }
+    
+    @Override
+    public LifelineDelayedEffect copy() {
+        return new LifelineDelayedEffect(this);
+    }
+    
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Cards cards = (Cards) this.getValue("LifelineCards");

```

Comment:
The reason why this works is because it dereferences the Permanent into an id when you put it into a 'Cards' object and then pulls it back out as the Card. You shouldn't be using a permanent as your point of comparison.